pkg/operator/controller/ingress/status_test.go

```diff
@@ -108,6 +111,131 @@ func cond(t string, status operatorv1.ConditionStatus, reason string, lt time.Ti
+				cond(operatorv1.DNSManagedIngressConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Minute*-3)),
+				cond(operatorv1.DNSReadyIngressConditionType, operatorv1.ConditionFalse, "", clock.Now().Add(time.Second*-15)),
+			},
+			expectedStatus: operatorv1.ConditionFalse,
+			expectRequeue:  true,
+		},
+		{
+			name: "DNS failing >30s",
+			conditions: []operatorv1.OperatorCondition{
+				cond(operatorv1.LoadBalancerManagedIngressConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Minute*-3)),
+				cond(operatorv1.LoadBalancerReadyIngressConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Second*-120)),
+				cond(operatorv1.DNSManagedIngressConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Minute*-3)),
+				cond(operatorv1.DNSReadyIngressConditionType, operatorv1.ConditionFalse, "", clock.Now().Add(time.Minute*-2)),
+			},
+			expectedStatus: operatorv1.ConditionTrue,
+			expectRequeue:  true,
+		},
+		{
+			name: "DNS not ready and deployment degraded",
+			conditions: []operatorv1.OperatorCondition{
+				cond(iov1.IngressControllerAdmittedConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Hour*-1)),
+				cond(iov1.IngressControllerDeploymentDegradedConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Hour*-1)),
+				cond(operatorv1.LoadBalancerManagedIngressConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Hour*-1)),
+				cond(operatorv1.LoadBalancerReadyIngressConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Hour*-1)),
+				cond(operatorv1.DNSManagedIngressConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Hour*-1)),
+				cond(operatorv1.DNSReadyIngressConditionType, operatorv1.ConditionFalse, "", clock.Now().Add(time.Hour*-1)),
+			},
+			expectedStatus: operatorv1.ConditionTrue,
+			expectRequeue:  true,
+		},
+		{
+			name: "admitted, DNS, LB, and deployment OK",
+			conditions: []operatorv1.OperatorCondition{
+				cond(iov1.IngressControllerAdmittedConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Hour*-1)),
+				cond(iov1.IngressControllerDeploymentDegradedConditionType, operatorv1.ConditionFalse, "", clock.Now().Add(time.Hour*-1)),
+				cond(operatorv1.LoadBalancerManagedIngressConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Hour*-1)),
+				cond(operatorv1.LoadBalancerReadyIngressConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Hour*-1)),
+				cond(operatorv1.DNSManagedIngressConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Hour*-1)),
+				cond(operatorv1.DNSReadyIngressConditionType, operatorv1.ConditionTrue, "", clock.Now().Add(time.Hour*-1)),
+			},
+			expectedStatus: operatorv1.ConditionFalse,
+			expectRequeue:  false,
+		},
+	}
+
+	for _, test := range tests {
+		actual, err := computeIngressDegradedCondition(test.conditions)
+		if actual.Status != test.expectedStatus {
```

Comment:
Do we want to check that `actual` can never be `nil` in the face of a non-`nil` error?

Reply:
I've added a check for non-retryable errors.